File: vripper-server/src/main/java/tn/mnlr/vripper/EventListenerBean.java
package tn.mnlr.vripper;

import lombok.Getter;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.event.ContextRefreshedEvent;
import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;
import tn.mnlr.vripper.jpa.repositories.IPostRepository;
import tn.mnlr.vripper.services.DataService;

@Component
public class EventListenerBean {

  @Getter private static boolean init = false;

  private final IPostRepository postRepository;
  private final DataService dataService;

  @Autowired
  public EventListenerBean(IPostRepository postRepository, DataService dataService) {
    this.postRepository = postRepository;
    this.dataService = dataService;
  }

  @EventListener
  public void onApplicationEvent(ContextRefreshedEvent event) {
    postRepository.setDownloadingToStopped();
    dataService.sortPostsByRank();
    init = true;
  }
}

File: vripper-core/src/main/kotlin/me/mnlr/vripper/listeners/EventListenerBean.kt
package me.mnlr.vripper.listeners

import org.springframework.context.event.ContextRefreshedEvent
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component
import me.mnlr.vripper.repositories.PostDownloadStateRepository
import me.mnlr.vripper.services.DataTransaction

@Component
class EventListenerBean(
    private val postDownloadStateRepository: PostDownloadStateRepository,
    private val dataTransaction: DataTransaction
) {
    @EventListener
    fun onApplicationEvent(event: ContextRefreshedEvent?) {
        postDownloadStateRepository.setDownloadingToStopped()
        dataTransaction.sortPostsByRank()
    }
}
